### PR TITLE
Blocks/Renderer: Ensure images have `srcset` and `sizes` attributes

### LINKED ISSUE #7983: Blocks/Renderer: Ensure images have `srcset` and `sizes` attributes
## Feature Description

Off the back of https://github.com/google/web-stories-wp/issues/7981. In WordPress 4.4, responsive images [were added](https://make.wordpress.org/core/2015/11/10/responsive-images-in-wordpress-4-4/). This automatically adds source set attributes to rendered image tags in content and templates. This is a performance win. However, how we render image tag, these attributes are not added. 

## Additional Context

- Function [wp_calculate_image_srcset](https://developer.wordpress.org/reference/functions/wp_calculate_image_srcset/)
- Function [wp_calculate_image_sizes](https://developer.wordpress.org/reference/functions/wp_calculate_image_sizes/)

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Add `srcset` and `sizes` attributes to all images that are rendered in web stories. 

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #7983

### CODE REVIEW

File: includes/Renderer/Stories/Renderer.php [hunk 1]
```diff
@@ -590,6 +590,8 @@ protected function render_story_with_poster() {
 						alt="<?php echo esc_attr( $story->get_title() ); ?>"
 						width="<?php echo absint( $this->width ); ?>"
 						height="<?php echo absint( $this->height ); ?>"
+						srcset="<?php echo esc_attr( $story->get_publisher_logo_srcset() ); ?>"
+						sizes="<?php echo esc_attr( $story->get_publisher_logo_sizes() ); ?>"
```
Comment (swissspidy) [inline includes/Renderer/Stories/Renderer.php]: `publisher_logo` -> `poster`

File: includes/Model/Story.php [hunk 2]
```diff
@@ -151,10 +166,17 @@ public function load_from_post( $_post ): bool {
...
 
 		if ( 0 !== $thumbnail_id ) {
-			$poster_url = wp_get_attachment_image_url( $thumbnail_id, Image_Sizes::POSTER_PORTRAIT_IMAGE_SIZE );
-
-			if ( $poster_url ) {
-				$this->poster_portrait = $poster_url;
+			$poster_src = wp_get_attachment_image_src( $thumbnail_id, Image_Sizes::POSTER_PORTRAIT_IMAGE_SIZE );
+
+			if ( $poster_src ) {
+				list ( $poster_url, $width, $height ) = $poster_src;
```
Comment (timarney) [inline includes/Model/Story.php]: Should we be using  `[$a, $b, $c] = $array;` syntax here?

**Comment by spacedmonkey** [comment]
This is a draft PR, reviews were not requested.

File: includes/Model/Story.php [hunk 3]
```diff
@@ -83,6 +83,21 @@ class Story {
...
 	protected $publisher_logo;
 
+	/**
+	 * Poster source set sizes.
+	 *
+	 * @var string
+	 */
+	protected $poster_sizes = '';
+
+
```
Comment (swissspidy) [inline includes/Model/Story.php]:
```suggestion

```

File: includes/Model/Story.php [hunk 4]
```diff
@@ -178,6 +201,30 @@ public function load_from_post( $_post ): bool {
...
+	 *
+	 * @since 1.18.0
+	 *
+	 * @return string
+	 */
+	public function get_poster_sizes(): string {
+		return (string) $this->poster_sizes;
+	}
+
+
```
Comment (swissspidy) [inline includes/Model/Story.php]:
```suggestion

```

File: includes/Model/Story.php [hunk 5]
```diff
@@ -178,6 +201,30 @@ public function load_from_post( $_post ): bool {
...
+	 *
+	 * @since 1.18.0
+	 *
+	 * @return string
+	 */
+	public function get_poster_srcset(): string {
+		return (string) $this->poster_srcset;
+	}
+
+
```
Comment (swissspidy) [inline includes/Model/Story.php]:
```suggestion

```

**Review by swissspidy — APPROVED** [review]

File: includes/Model/Story.php [hunk 6]
```diff
@@ -83,6 +83,19 @@ class Story {
...
+	 *
+	 * @var string
+	 */
+	protected $poster_sizes = '';
+	/**
+	 * Poster source set.
+	 *
+	 * @var string
+	 */
+	protected $poster_srcset = '';
```
Comment (swissspidy) [inline includes/Model/Story.php]: Wait I meant to only remove only 1 newline of the double newlines you had here everywhere, but now all newlines were removed 🤔

Comment (spacedmonkey) [reply includes/Model/Story.php]: I just applied all the suggested changes.

Comment (spacedmonkey) [reply includes/Model/Story.php]: Fixed.

**Comment by spacedmonkey** [comment]
Does this needed to be QAed?

**Comment by swissspidy** [comment]
Can it be QAed? The PR description is not indicative over whether there is any user-facing change or testing instructions.

**Comment by spacedmonkey** [comment]
> Can it be QAed? The PR description is not indicative over whether there is any user-facing change or testing instructions.

I guess it can be QAed, as in, checking nothing is broken. But the real check if creating blocks and seeing the attributes. Which I have done. Does this need to repeated by QA?

**Comment by swissspidy** [comment]
Our rule of thumb has always been that we can skip QA for smaller changes (especially non-user-facing ones) that have already been vetted in code review. This clearly applies here. In other words, feel free to merge.

**Comment by rodrigofglima** [comment]
Hi! I have a question: I'm using an extra WordPress plugin that generates the crops with the webp format (when I upload the images in jpg), because my portal hosting disk space is limited and I can't ask for more space this year (2023 only), and I want to avoid using CDN for now. In this case, may this change have an possible conflict, because the extra crops are in a different file extension/format?

I also re-enabled the Jetpack system that generates the extra images, but sometimes I disable it because the quality of the images is poor in the format.

**Comment by swissspidy** [comment]
@rodrigofglima The file type doesn't matter.

Please note that if you have general web development or WordPress questions, this isn't the right place. And if you have questions about the Web Stories plugin, you should use our support forums instead. Thanks!

**Pull request merged**